cloud/amazon/iam.py

```diff
@@ -0,0 +1,539 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: iam
+short_description: Manage IAM users, groups, roles and keys
+description:
+     - Allows for the management of IAM users, groups, roles and access keys.
+version_added: "1.9"
+options:
+  iam_type:
+    description:
+      - Type of IAM resource
+    required: "true"
+    default: null
+    choices: [ "user", "group", "role"]
+    aliases: []
+  name:
+    description: name of IAM resource to create or identify
+    required: "true"
+    aliases: []
+  new_name:
+    description: when state is update, will replace name with new_name on IAM resource
+    required: "false"
+    aliases: []
+  new_path:
+    description: when state is update, will replace the path with new_path on the IAM resource
+    required: "false"
+    aliases: []
+  state:
+    description:
+      - Whether to create, delete or update the IAM resource. Note, roles cannot be updated.
+    required: "true"
+    default: null
+    choices: [ "present", "absent", "update" ]
+    aliases: []
+  path:
+    description:
+      - when creating or updating, specify the desired path of the resource
+    required: "false"
+    default: "/"
+    aliases: []
+  access_key_state:
+    description:
+      - when type is user: creates, removes, deactivates or activates a user's access key(s). Note that actions apply only to keys specified.
+    required: "false"
+    default: null
+    choices: ["create", "remove", "Active", "Inactive"]
+    aliases: []
+  access_key_ids:
+    description:
+      - a list of the keys that you want impacted by the access_key_state paramter.
+  groups:
+    description:
+      - A list of groups the user should belong to. When update, will gracefully the remove groups not listed.
```

Comment:
Ah! I was looking for 'the the'. Thanks. Fixed. Will update the P/R within the hour with this change and the stuff mentioned below.